app/models/schedule.rb

```diff
@@ -2,13 +2,22 @@ class Schedule < ApplicationRecord
   belongs_to :user
   has_many :slots, inverse_of: :schedule, dependent: :destroy
   accepts_nested_attributes_for :slots, allow_destroy: true
-  scope :by_from, -> { order(:from) }
+  scope :by_start_at, -> { order(:start_at) }
 
-  validates :from, presence: true
-  validate :from_must_be_more_than_six_weeks_in_the_future, unless: :first_schedule?
+  validates :start_at, presence: true
+  validate :start_at_must_be_more_than_six_weeks_in_the_future, unless: :first_schedule?
+
+  def end_at
+    self[:end_at] - 1.day if self[:end_at]
```

Comment:
We can't actually, because of the way the code is now. There is no :end_at accessor defined. I've tried defining one and it's always nil even if self[:end_at] is not. Bit confused about how rails sets those ivars.